ansibulled/cli/ansibulled_changelog.py

```diff
@@ -234,8 +237,8 @@ def command_generate(args: Any) -> None:
         print('Cannot create changelog when not at least one release has been added.')
         sys.exit(2)
     if reload_plugins:
-        plugins = load_plugins(paths=paths, version=changes.latest_version,
-                               force_reload=reload_plugins)
+        plugins: Optional[List[PluginDescription]] = load_plugins(
+            paths=paths, version=changes.latest_version, force_reload=reload_plugins)
```

Comment:
A return value of Optional doesn't seem right.  I think the API should do one of the following:
* raise an exception if nothing is found
* return an empty container if it returns a container type

I'm new to using type checks for things so I'm not sure how it's best to use them in conjunction with code like this.  Perhaps for new code, it shouldn't be added?  But for existing code like this it's okay to add Optional and flag it so that we know it's an API we need to go back and review?  (Take advantage of pylint and use `# TODO: Fix API so it returns and empty list instead of None`?  Write a script to find places where we use `Optional[List]`?  So many possibilities but I don't know which is going to be the best to work with down the road ;-)

Reply:
The return value is `List[PluginDescription]`. The type of the variable `plugins` is `Optional[List[PluginDescription]]`, though, and if it is not set explicitly to it, the line `plugins = None` slightly below will produce an error.

Reply:
I just noticed that it actually works to define the type before using the variable, in this case before the `if`:
```.py
    plugins: Optional[List[PluginDescription]]
    if reload_plugins:
        plugins = load_plugins(...)
    else:
        plugins = None
```
(I tested that earlier and it resulted in an error, probably because of something else, and thought it wouldn't work like this...)

Reply:
<nod> the common way to write that is probably:

``` python
    plugins: Optional[List[PluginDescription]] = None
    if reload_plugins:
        plugins = load_plugins(...)
```